src/dune_engine/cached_digest.ml

```diff
@@ -181,68 +181,63 @@ let set path digest =
 ;;
 
 module Digest_result = struct
-  type t =
-    | Ok of Digest.t
-    | No_such_file
-    | Broken_symlink
-    | Cyclic_symlink
-    | Unexpected_kind of File_kind.t
-    | Unix_error of Unix_error.Detailed.t
-    | Error of exn
-
-  let equal x y =
-    match x, y with
-    | Ok x, Ok y -> Digest.equal x y
-    | Ok _, _ | _, Ok _ -> false
-    | No_such_file, No_such_file -> true
-    | No_such_file, _ | _, No_such_file -> false
-    | Broken_symlink, Broken_symlink -> true
-    | Broken_symlink, _ | _, Broken_symlink -> false
-    | Cyclic_symlink, Cyclic_symlink -> true
-    | Cyclic_symlink, _ | _, Cyclic_symlink -> false
-    | Unexpected_kind x, Unexpected_kind y -> File_kind.equal x y
-    | Unexpected_kind _, _ | _, Unexpected_kind _ -> false
-    | Unix_error x, Unix_error y ->
-      Tuple.T3.equal Unix_error.equal String.equal String.equal x y
-    | Unix_error _, _ | _, Unix_error _ -> false
-    | Error x, Error y ->
-      (* Falling back to polymorphic equality check seems OK for this rare case.
-         We could also just return [false] but that would break the reflexivity
-         of the equality check, which doesn't seem nice. *)
-      x = y
-  ;;
+  module Error = struct
+    type t =
+      | No_such_file
+      | Broken_symlink
+      | Cyclic_symlink
+      | Unexpected_kind of File_kind.t
+      | Unix_error of Unix_error.Detailed.t
+      | Error of exn
```

Comment:
Now that it's inside the `Error` module, I'd prefer to call this variant `Unknown` or `Unrecognized`.